presto-hudi/src/main/java/com/facebook/presto/hudi/HudiPartitionManager.java

```diff
@@ -78,6 +84,56 @@ public List<String> getEffectivePartitions(
                 partitionPredicate.put(partitionColumn, Domain.all(column.getHiveType().getType(typeManager)));
             }
         }
-        return metastore.getPartitionNamesByFilter(metastoreContext, schemaName, tableName, partitionPredicate);
+        List<String> partitionNames = metastore.getPartitionNamesByFilter(metastoreContext, schemaTableName.getSchemaName(), schemaTableName.getTableName(), partitionPredicate);
+        List<Type> partitionTypes = partitionColumns.stream()
+                .map(column -> typeManager.getType(column.getType().getTypeSignature()))
+                .collect(toList());
+
+        return partitionNames.stream()
+                // Apply extra filters which could not be done by getPartitionNamesByFilter, similar to filtering in HivePartitionManager#getPartitionsIterator
+                .filter(partitionName -> parseValuesAndFilterPartition(
+                        partitionName,
+                        hudiColumnHandles,
+                        partitionTypes,
+                        constraintSummary))
+                .collect(toList());
+    }
+
+    private boolean parseValuesAndFilterPartition(
+            String partitionId,
+            List<HudiColumnHandle> partitionColumns,
+            List<Type> partitionColumnTypes,
+            TupleDomain<ColumnHandle> constraintSummary)
+    {
+        if (constraintSummary.isNone()) {
```

Comment:
This is just a sanity check. We should never land in this block because in `HudiMetadata.getTableLayouts` we always get the constrain with data and partition columns.